RunCat365/StartupManager.cs

```diff
@@ -0,0 +1,145 @@
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+using Microsoft.Win32;
+using Windows.ApplicationModel;
+
+namespace RunCat365
+{
+    internal static class StartupManager
+    {
+
+        private static StartupTask? startupTask;
+
+        public static bool GetStartup()
+        {
+            if (IsRunningAsPackaged())
+            {
+                return GetStartupAsync_Packed().Result;
+            }
+            else
+            {
+                return GetStartup_NotPacked();
+            }
+        }
+
+        public static bool SetStartup(bool isChecked)
+        {
+            if (IsRunningAsPackaged())
+            {
+                return SetStartupAsync_Packed(isChecked).Result;
+            }
+            else
+            {
+                return SetStartup_NotPacked(isChecked);
+            }
+        }
+
+        private static async Task<bool> GetStartupAsync_Packed()
+        {
+            if (startupTask is null) startupTask = await StartupTask.GetAsync("RunCatStartup");
+            if (startupTask is null) return false;
+            if (startupTask.State == StartupTaskState.Enabled) return true;
+            return false;
+        }
+
+        private static bool GetStartup_NotPacked()
+        {
+            var keyName = @"Software\Microsoft\Windows\CurrentVersion\Run";
+            using var rKey = Registry.CurrentUser.OpenSubKey(keyName);
+            if (rKey is null) return false;
+            var value = (rKey.GetValue(Application.ProductName) is not null);
+            rKey.Close();
+            return value;
+        }
+
+        private static async Task<bool> SetStartupAsync_Packed(bool isChecked)
+        {
+            var active = !isChecked;
+            var changeCheck = false;
+            if (startupTask is null) startupTask = await StartupTask.GetAsync("RunCatStartup");
+            if (active)
+            {
+                switch (startupTask.State)
+                {
+                    case StartupTaskState.Enabled:
+                        changeCheck = true;
+                        break;
+                    case StartupTaskState.Disabled:
+                        StartupTaskState newState = await startupTask.RequestEnableAsync();
+                        if (newState == StartupTaskState.Enabled)
+                        {
+                            changeCheck = true;
+                        }
+                        else
+                        {
+                            MessageBox.Show("Launch at Startup could not be activated.", "Warning", MessageBoxButtons.OK, MessageBoxIcon.Warning);
```

Comment:
Instead of calling `MessageBox.Show()` here, it would be better to throw an error and handle it in the `ContextMenuManager` that uses `SetStartup`. 
Since `ContextMenuManager` deals with UI-related operations, it makes more sense there.